cpp/src/db/db_object.cu

```diff
@@ -216,19 +194,22 @@ idx_t db_column_index<idx_t>::getIndirectionSize() {
 }
 
 template<typename idx_t>
-std::string db_column_index<idx_t>::toString(){
+std::string db_column_index<idx_t>::toString() {
   std::stringstream ss;
   ss << "db_column_index:\n";
   ss << "Offsets: ";
-  idx_t* hostOffsets = (idx_t*)malloc(sizeof(idx_t) * offsets_size);
-  cudaMemcpy(hostOffsets, offsets, sizeof(idx_t) * offsets_size, cudaMemcpyDefault);
+  idx_t* hostOffsets = (idx_t*) malloc(sizeof(idx_t) * offsets_size);
```

Comment:
As per today's discussion, we are trying to get rid of `malloc`

Reply:
The malloc's used here are inside of a debugging-only toString() method. They aren't in code that will get exposed to end users. Brad has said it's okay to leave it as is.

Reply:
My biggest concern is, this will appear in search, and it is very difficult to figure out which malloc is benign and which is not. This will make finding problematic mallocs very difficult as we need to inspect every search appearance of malloc case by case. If this is not too much additional work, it is great if you can avoid this.

Reply:
thrust::host_vector<idx_t> host_offsets = offsets;
for (idx_t i = 0; i < offsets_size; ++i) {
  ss << host_offsets[i] << " ";
}
works and simpler.

And if you don't care about performance (if this is debugging only),

offsets[i] in the host side works correctly (offsets here is rmm::device_vector which is a wrapper for thrust::device_vector, right?).

Reply:
I think something like the following will be cleaner.
```
  ss << "Offsets:";
  for (auto it = offsets.cbegin(); it != offsets.cend(); ++it) {
    ss << " " << *it;
  }
  ss << "\n";
  
  ss << "Indirection:";
  for (auto it = indirection.cbegin(); it != indirection.cend(); ++it) {
    ss << " " << *it;
  }
  ss << "\n";
```